lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1232,12 +1231,35 @@ private synchronized void onConnected(ProtocolMessage message) {
             requestState(transport, new StateIndication(ConnectionState.failed, e.errorInfo));
             return;
         }
-
         /* indicated connected currentState */
         setSuspendTime();
         requestState(new StateIndication(ConnectionState.connected, error));
     }
 
+    /**
+     * Send all pending messages which are in the queue.
+     * Remove them from the queue once they are sent successfully
+     * Spec: RTN19a
+     * @param resetMessageSerial
+     */
+    private void sendPendingQueueMessages(boolean resetMessageSerial) {
```

Comment:
Personally I would change `resetStartSerial()` so that it returns the next unassigned `msgSerial`.

Then I would remove this boolean `resetMessageSerial` argument from here.

The calling code above (`RTN15c3` case) should call `resetStartSerial()` directly, and use the result to set `this.msgSerial`.